### PR TITLE
Parameterized Address Layout (Address 3 /Address 4) via SysConf

### PR DESCRIPTION
**Pull request by Chemssou**

### CODE REVIEW

File: backend/de.metas.ui.web.base/src/main/java/de/metas/ui/web/address/AddressDescriptorFactory.java [hunk 1]
```diff
@@ -108,17 +125,25 @@ private DocumentEntityDescriptor createAddressEntityDescriptor()
...
-				.setDataBinding(new AddressFieldBinding(IAddressModel.COLUMNNAME_Address3, false, I_C_Location::getAddress3, AddressFieldBinding::writeValue_Address3)));
-		//
-		addressDescriptor.addField(buildFieldDescriptor(IAddressModel.COLUMNNAME_Address4)
-				.setValueClass(String.class)
-				.setWidgetType(DocumentFieldWidgetType.Text)
-				.setDataBinding(new AddressFieldBinding(IAddressModel.COLUMNNAME_Address4, false, I_C_Location::getAddress4, AddressFieldBinding::writeValue_Address4)));
-		//
+
+		final boolean allowAddress3 = isShowAddress3();
+		if(allowAddress3)
```
Comment (teosarca) [inline backend/de.metas.ui.web.base/src/main/java/de/metas/ui/web/address/AddressDescriptorFactory.java:130]: cosmetics: 

just inline this. there is no need for a local var when it's the only place where is used.

i.e. change it to
```
if(isShowAddress3()) { 
....

```

File: backend/de.metas.ui.web.base/src/main/java/de/metas/ui/web/address/AddressDescriptorFactory.java [hunk 2]
```diff
@@ -108,17 +125,25 @@ private DocumentEntityDescriptor createAddressEntityDescriptor()
...
+		{
+			addressDescriptor.addField(buildFieldDescriptor(IAddressModel.COLUMNNAME_Address3)
+					.setValueClass(String.class)
+					.setWidgetType(DocumentFieldWidgetType.Text)
+					.setDataBinding(new AddressFieldBinding(IAddressModel.COLUMNNAME_Address3, false, I_C_Location::getAddress3, AddressFieldBinding::writeValue_Address3)));
+		}
+
+		final boolean showAddress4 = isShowAddress4();
+		if(showAddress4)
```
Comment (teosarca) [inline backend/de.metas.ui.web.base/src/main/java/de/metas/ui/web/address/AddressDescriptorFactory.java:139]: pls inline

File: backend/de.metas.ui.web.base/src/main/java/de/metas/ui/web/address/AddressDescriptorFactory.java (hunk 1 above)
Comment (Chemssou) [reply backend/de.metas.ui.web.base/src/main/java/de/metas/ui/web/address/AddressDescriptorFactory.java]: Done

File: backend/de.metas.ui.web.base/src/main/java/de/metas/ui/web/address/AddressDescriptorFactory.java (hunk 2 above)
Comment (Chemssou) [reply backend/de.metas.ui.web.base/src/main/java/de/metas/ui/web/address/AddressDescriptorFactory.java]: @teosarca  Done

**Pull request merged**